arviz/data/io_pymc3.py

```diff
@@ -153,18 +155,25 @@ def _extract_log_likelihood(self):
         """Compute log likelihood of each observation."""
         # If we have predictions, then we have a thinned trace which does not
         # support extracting a log likelihood.
-        cached = [(var, var.logp_elemwise) for var in self.model.observed_RVs]
-        log_likelihood_dict = {}
+        if isinstance(self.log_likelihood, list):
+            cached = [
+                (var, var.logp_elemwise)
+                for var in self.model.observed_RVs
+                if var in self.log_likelihood
+            ]
+        else:
+            cached = [(var, var.logp_elemwise) for var in self.model.observed_RVs]
+        log_likelihood_dict = utils._sampledict(  # pylint: disable=protected-access
+            len(self.trace.chains)
```

Comment:
Actually, what this does is pass the length for the first dim for the array for each variable (key). Since, each key is looped for total chains, I guess the shape of array for the key would be ```len(self.trace.chains) + val.shape```? Do you want to initialize the whole dict in one go?